packages/web-components/src/components/accordion/Accordion.ts

```diff
@@ -0,0 +1,179 @@
+    return { heading, content, closeButton };
+  }
+
+  toggleOpen = () => {
+    this.isOpen = !this.isOpen;
+  }
+
+  hasCloseButton = () =>
+    !!this.closeButton;
+
+  protected willUpdate(_changedProperties: PropertyValueMap<unknown> | Map<PropertyKey, unknown>): void {
+    // TODO: Figure out how this side-effect-ish className would be better handled
+    //       e.g. best practices for non-ortogonal stuff in lit library.
+    const card = this.card && 'card'
+    const border = this.card && this.border && 'border'
+    const isOpen = this.isOpen && 'isOpen'
+    this.className = classNames('accordion', card, border, isOpen, this.size)
+    return super.willUpdate(_changedProperties)
+  }
+
+  render () {
+    const handleKeyPress = (e) => {
+      if (e.key === 'Enter' || e.key === ' ') {
+        this.toggleOpen(); // FIXME
+      }
+    }
+
+    return html`
+      <div class='accordionHeader'>
+        <div
+          id=${this.identifiers.heading}
+          aria-level=${this.headingLevel}
+          role="heading">
+
+          <div
+            role='button'
+            class='headingContainer'
+            tabIndex=0
+            aria-labelledby=${this.identifiers.heading}
+            aria-expanded=${this.isOpen}
+            @keypress=${handleKeyPress}
+            @click=${this.toggleOpen}>
+
+            <span class='label'>${this.heading}</span>
+            ${this.renderIcon()}
+          </div>
+
+        </div>
+      </div>
+
+      <div
+        id=${this.identifiers.content}
+        style="display: ${this.isOpen ? 'block' : 'none'}"
+        role="region"
+        class='accordionContent'
+        aria-labelledby=${this.identifiers.heading}>
+        <slot></slot>
+        ${this.renderCloseButton()}
+      </div>
+    </div>
+    `;
+  }
+
+  private renderIcon () {
+    return html`<svg
+      xmlns="http://www.w3.org/2000/svg"
+      viewBox="0 0 24 24"
+      role="img">
+      <g fill="none" fillRule="evenodd">
+        <path d="M0 24h24V0H0z" />
+        <path fill="currentColor" d="M12 11.5l5 5 1.5-1.5L12 8.5 5.5 15 7 16.5z" />
+      </g>
+    </svg>`;
+  }
+
+  private renderCloseButton () {
+    if (!this.hasCloseButton)
+      return '';
+
+    const closeMessage  = getCloseMessage(this.language);
+    const closeLabel    = classNames(closeMessage, this.heading)
+    const className     = classNames('closeButton', this.closeButtonClassName)
+    // TODO: .iconRight={<IconAngleUp aria-hidden size="xs" className={styles.accordionButtonIcon} />}
+
+    const handleKeyPress = (e) => {
+      if (e.key === ' ') {
+        this.toggleOpen(); // FIXME
+      }
+    }
+
+    return html`
+      <hds-button
+        class=${className}
+        variant='supplementary'
+        aria-label=${closeLabel}
+        theme='black'
+        size='small'
+        @keypress=${handleKeyPress}
+        @click=${this.toggleOpen}>
+        ${closeMessage}
+      </hds-button>
```

Comment:
Button is missing chevron but I think we discussed that we will go without it for now because it would need more work.